metaflow/datatools/s3.py

```diff
@@ -575,8 +575,7 @@ def _read_many_files(self, op, prefixes, **options):
                                                      **options)
             if stderr:
                 raise MetaflowS3Exception("Getting S3 files failed.\n"\
-                                          "First prefix requested: %s\n"\
-                                          "Error: %s" % (prefixes[0], stderr))
+                                          "Error: %s" % (stderr,))
```

Comment:
You can instead do `prefixes = list(prefixes)` after L:566

Reply:
Updated. 😀 

I didn't spot the same pattern in the method below (`_put_many_files`). I initially thought the use of generators might have been some kind of performance/memory optimisation.